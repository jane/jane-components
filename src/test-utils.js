// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'
import theme from './theme'
import { ThemeProvider } from 'styled-components'

export const create = (C: React.Node): string =>
  renderer.create(<ThemeProvider theme={theme}>{C}</ThemeProvider>).toJSON()
