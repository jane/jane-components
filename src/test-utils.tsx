import * as React from 'react'
import renderer from 'react-test-renderer'
import theme from './theme'
import { ThemeProvider } from 'styled-components'

type RendererJson = {
  type: string
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  props: { [propName: string]: any }
  children: null | Array<RendererJson>
}

export const create = (C: React.ReactElement): RendererJson | null =>
  renderer.create(<ThemeProvider theme={theme}>{C}</ThemeProvider>).toJSON()
