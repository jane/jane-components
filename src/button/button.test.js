// @flow

/* eslint-env jest */

import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from '../test-utils'
import Button from './button'

describe('button', (): void => {
  it('renders', (): void => {
    expect(create(<Button />)).toMatchSnapshot()
  })

  it('has onClick when passed', (): void => {
    const handleClick = jest.fn()

    const button = shallow(
      <Button onClick={handleClick}>
        <i />
      </Button>
    )

    expect(typeof button.prop('onClick')).toBe('function')
    button.simulate('click')
    expect(handleClick).toHaveBeenCalled()
  })

  it('has box-shadow with type', (): void => {
    expect(create(<Button type="warning">foo</Button>)).toMatchSnapshot()
  })
})
