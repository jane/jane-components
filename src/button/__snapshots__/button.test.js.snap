// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button has box-shadow with type 1`] = `
.c0 {
  white-space: nowrap;
  background-color: #F5941C;
  border-width: 0;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 16px;
  line-height: 19px;
  padding: 12px 28px;
  box-shadow: 0 0 0 4px #F5941C;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 4px;
  overflow: hidden;
  opacity: 1;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus,
.c0:active {
  outline: none;
  background-color: #DB8419;
  box-shadow: 0 0 0 4px #DB8419;
}

.c0:active {
  box-shadow: 0 0 0 2px #DB8419;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: 0.16s;
  transition-duration: 0.16s;
  -webkit-transition-delay: 0s;
  transition-delay: 0s;
  -webkit-transition-timing: ease-out;
  transition-timing: ease-out;
}

<button
  className="c0"
  type="warning"
>
  foo
</button>
`;

exports[`button renders 1`] = `
.c0 {
  white-space: nowrap;
  border-width: 0;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 16px;
  line-height: 19px;
  padding: 12px 28px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 4px;
  overflow: hidden;
  opacity: 1;
  text-transform: uppercase;
}

.c0:hover,
.c0:focus,
.c0:active {
  outline: none;
}

.c0:active {
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: 0.16s;
  transition-duration: 0.16s;
  -webkit-transition-delay: 0s;
  transition-delay: 0s;
  -webkit-transition-timing: ease-out;
  transition-timing: ease-out;
}

<button
  className="c0"
/>
`;
